hoomd/Saru.h

```diff
@@ -597,53 +418,41 @@ HOSTDEVICE inline double Saru::s(double a, double b)
     }
 
 /*!
- * \tparam steps Number of steps to advance.
- */
-template <unsigned int steps>
-HOSTDEVICE inline void Saru::advanceLCG()
-    {
-    state.x = CTpow<LCGA,steps>::value*state.x+LCGC*CTpowseries<LCGA,steps>::value;
-    }
-
-/*!
- * \param x
+ * \param sigma Standard deviation
+ * \param mu Mean
+ *
+ * \returns Normally distributed random variable with mean *mu* and standard deviation *sigma*.
  *
- * \tparam offset
- * \tparam delta
- * \tparam modulus
- * \tparam steps Number of steps to advance.
+ * \post The state of the generator is advanced one step.
  */
-template <unsigned int offset, unsigned int delta, unsigned int modulus, unsigned int steps>
-HOSTDEVICE inline unsigned int Saru::advanceAnyWeyl(unsigned int x)
+HOSTDEVICE inline float Saru::normal(float sigma, float mu)
     {
-    const unsigned int fullDelta=CTmultmod<delta, steps%modulus, modulus>::value;
-    /* the runtime code boils down to this single constant-filled line. */
-    return x+((x-offset>modulus-fullDelta) ? fullDelta-modulus : fullDelta);
+    r123::Philox4x32 rng;
+    r123::Philox4x32::ctr_type u = rng(m_ctr, m_key);
+    m_ctr[0] += 1;
+    float2 n = r123::boxmuller(u[0], u[1]);
+    return n.x * sigma + mu;
+    // note: If there is a need, we could add an API that returns two normally distributed numbers for no extra cost
```

Comment:
How should that API look: `float2 Saru::normal(float2 sigma, float2 mu)`?

Reply:
That would work but it is a little clunky if you want two values with the same `sigma` and `mu`, which is the common use case (will need `sigma*make_scalar2(1,1)` everywhere). This is also not ideal, but would still work with overloading:
```
void Saru::normal(float sigma, float mu, float2& vals);
void Saru::normal(double sigma, double mu, double2& vals);
```
Not sure if one is better than the other from a usage / performance perspective. Otherwise, templates would be needed.